searchcore/src/vespa/searchcore/proton/matching/matcher.cpp

```diff
@@ -303,10 +283,20 @@ Matcher::match(const SearchRequest &request,
         }
         size_t covered = (spaceEstimate *  numActiveLids) / totalSpace;
         LOG(debug, "covered = %zd", covered);
+
+        SearchReply::Coverage & coverage = reply->coverage;
         coverage.setActive(numActiveLids);
         //TODO this should be calculated with ClusterState calculator.
         coverage.setSoonActive(numActiveLids);
         coverage.setCovered(covered);
+        if (wasLimited) {
+            coverage.degradeMatchPhase();
+            LOG(debug, "was limited, degraded from match phase");
+        }
+        if (my_stats.softDoomed()) {
+            coverage.degradeTimeout();
+            coverage.setCovered(my_stats.docsCovered());
```

Comment:
need to convert from searched docid space to actual document count. maybe rename to docidSpaceCovered in statd and always use this value for estimate (above)?

Reply:
(not that using the stat above removed the need for +1 since docid 0 is already not covered by searched ranges)

Reply:
Done.